### PR TITLE
feature/domain-refactor

### PR DESCRIPTION
**Pull request by kane16**

### CODE REVIEW

**Comment by kane16** [comment]
## Code Review — `feature/domain-refactor`

**Scope:** +1,967 / −3,439 across 246 files. Three substantive changes under a lot of mechanical churn: (1) domain-first repackaging (~190 files are pure renames), (2) auth extraction to the external `pl.delukesoft:authplugin:1.0` dependency, (3) Liquibase-for-Mongo migration + Compose moving auth DB from MySQL to Postgres.

**Verified locally:** `./gradlew compileKotlin compileTestKotlin` → BUILD SUCCESSFUL. New validator unit tests are clean (BDD names, parameterized, good edge-case coverage).

### 🔴 Blockers
- **Secrets in `.env` (addressed in `319fefd` — untracked from git).** `.env` was tracked despite being in `.gitignore`, and contained a real `JWT_SECRET`, `AUTH_POSTGRES_PASSWORD`, `AUTH_PASSWORD`, and Mongo creds. It is now untracked. **Still required:** these values remain in git history — **rotate the JWT secret and every committed password**, and scrub history (`git filter-repo` / BFG) since the repo is on GitHub. Add a `.env.example` with placeholders.
- **Same `JWT_SECRET` hardcoded in `application-local.yml`.** The real token sits in source. Move to `${JWT_SECRET}` and rotate.

### 🟠 Major
- **No CI.** `gh pr checks` reports no checks. A 246-file refactor merging with zero automated verification is risky — consider a minimal GitHub Actions workflow running `./gradlew build`.
- **`.ai/mcp/mcp.json` committed empty** (0 bytes). Populate it or drop it.

### 🟡 Minor
- **Indentation/style on new `author/` files** (`AuthorConfiguration.kt`, `PortfolioAuthor.kt`) use 4-space indent; project standard is 2 (and `ResumeController.kt` in this PR uses 2). Several new files also lack a trailing newline. Run the formatter.
- **`PortfolioAuthorMapper.mapToApplicationAuthor`** calls `objectMapper.convertValue(author.additionalInfo)` unguarded — malformed `additionalInfo` throws a raw Jackson `IllegalArgumentException` that won't map cleanly via `ErrorAdvisor`. Wrap in a `LoggableResponseStatusException`.
- **`application-docker.yml`**: `auth.api.url: ${AUTH_API_URL:}` defaults to empty string → fails opaquely at runtime rather than at startup. Prefer no default (fail fast) or a real default.
- **`MongoLiquibaseConfig`**: dual flags (`spring.liquibase.enabled: false` + custom `app.liquibase.enabled`) plus `@Value` mongo username/password with no defaults are easy to misconfigure. A one-line comment on why both flags exist would help.

### ✅ Strengths
- Repackaging genuinely conforms to the documented domain-first structure; removes the `adapters`/`application` top-level packages the instructions flag as anti-patterns.
- Controllers keep conventions: `@AuthRequired`, `Authorization` header param, entry-point logging, no business logic.
- Good new validator unit tests with a shared `DateTimeUtils` / `ResumeValidatorTestBase`.
- Mapper/config separation respected (`PortfolioAuthorMapper` is a distinct `@Component`).

### Recommendation
**Request changes** — refactor is sound and compiles, but rotate + scrub the leaked secrets before merge. Minor items can ride along in the same cleanup.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: src/main/resources/application-docker.yml [hunk 1]
```diff
@@ -2,4 +2,4 @@ jwt:
   secret: ${JWT_SECRET}
 auth:
   api:
-    url: ${AUTH_API_URL}
\ No newline at end of file
+    url: ${AUTH_API_URL:}
```
Comment (kane16) [inline src/main/resources/application-docker.yml:5]: **Minor:** `${AUTH_API_URL:}` defaults to an empty string when the var is unset, so a misconfigured deployment fails opaquely at runtime instead of at startup. Prefer no default (fail fast) or a real default URL.

Also note: `application-local.yml:17` still hardcodes a real `JWT_SECRET` token in source — move it to `${JWT_SECRET}` and rotate it (it's in git history).

**Review by kane16 — COMMENTED** [review]
Inline review of the substantive (non-rename) changes. Compiles locally (`./gradlew compileKotlin compileTestKotlin` → BUILD SUCCESSFUL). The leaked-secrets blocker on `.env` is already addressed in 319fefd (untracked), but the values remain in git history and must be rotated + scrubbed.

File: src/main/kotlin/pl/delukesoft/portfolioserver/resume/author/PortfolioAuthorMapper.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import pl.delukesoft.authplugin.author.Author
+import pl.delukesoft.authplugin.author.AuthorMapper
+
+@Component
+class PortfolioAuthorMapper(
+    val objectMapper: ObjectMapper
+): AuthorMapper<PortfolioAuthor> {
+
+    override fun mapToApplicationAuthor(author: Author): PortfolioAuthor {
+        val additionalInfo = objectMapper.convertValue<PortfolioAuthorAdditionalInfo>(author.additionalInfo)
```
Comment (kane16) [inline src/main/kotlin/pl/delukesoft/portfolioserver/resume/author/PortfolioAuthorMapper.kt]: **Minor (error handling):** `objectMapper.convertValue(...)` is unguarded — malformed or missing `additionalInfo` throws a raw Jackson `IllegalArgumentException` that won't map cleanly through `ErrorAdvisor`. Per AI_AGENT_INSTRUCTIONS §4, wrap this in a domain `LoggableResponseStatusException` so the client gets a proper HTTP status.

File: src/main/kotlin/pl/delukesoft/portfolioserver/resume/author/AuthorConfiguration.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package pl.delukesoft.portfolioserver.resume.author
+
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import pl.delukesoft.authplugin.security.AuthContext
+
+@Configuration
+class AuthorConfiguration {
```
Comment (kane16) [inline src/main/kotlin/pl/delukesoft/portfolioserver/resume/author/AuthorConfiguration.kt:8]: **Style:** This file uses 4-space indentation, but the project standard is 2 spaces (AI_AGENT_INSTRUCTIONS §15 — and `ResumeController.kt` in this same PR uses 2). Lines 14-16 are also empty padding and the file has no trailing newline. Please run the formatter on the new `author/` files.

File: src/main/kotlin/pl/delukesoft/portfolioserver/platform/mongo/MongoLiquibaseConfig.kt [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import liquibase.Liquibase
+import liquibase.database.DatabaseFactory
+import liquibase.resource.ClassLoaderResourceAccessor
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+
+@Configuration
+@ConditionalOnProperty(prefix = "app.liquibase", name = ["enabled"], havingValue = "true", matchIfMissing = true)
```
Comment (kane16) [inline src/main/kotlin/pl/delukesoft/portfolioserver/platform/mongo/MongoLiquibaseConfig.kt]: **Minor (config clarity):** There are now two flags governing Liquibase — `spring.liquibase.enabled: false` (application.yml) and this custom `app.liquibase.enabled`. Combined with the `@Value` mongo username/password below (no defaults), this is easy to misconfigure. A one-line comment explaining why both flags coexist would help future readers.
